src/main/java/io/jenkins/plugins/ml/InterpreterManager.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 Loghi Perinpanayagam.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml;
+
+import org.apache.zeppelin.interpreter.InterpreterException;
+
+import java.io.IOException;
+
+/**
+ * Abstract class that can be subclassed for individual interpreter management
+ */
+
+public abstract class InterpreterManager {
+
+    abstract KernelInterpreter createInterpreter();
+    abstract void initiateInterpreter() throws InterpreterException;
+    abstract void closeInterpreter();
```

Comment:
Maybe the interpreter/kernel/etc with the `start`/`stop`, or `initiate`/`close` pattern could implement the `Closable` interface, so users of the API can do something like

try (final InterpreterManager interpreterManager = InterpreterManager.create()) {
  // use `interpreterManager`
}

// No need to close it, as Java 7+ takes care of that for us.

Just in case this fits with what you have in mind to implement in the future :+1:

Reply:
I thought that the user may want to close the interpreter in the middle. If we use `Closable`, will the user be able to close intentionally?

Reply:
It will work the same way. Only difference is that a user of the java API (most likely you and your code) can use try-with-resources - https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
I got that @kinow. I will implement `Closeable` there.

Reply:
Oh, it got more confused now. Let me start the IDE and have another look at the code later @Loghijiaha .

My bad. I thought this manager was going to be started/stopped, but looks like these methods are to start/stop the interpreter. Having `Closeable` here forces the implementations of a manager to have a `close()` method, but you have a `closeInterpreter`.

Let me just think for a bit and try the code in the IDE later (will do either today or in the next couple days)

Reply:
Nah, all good! Just had a look at the code with the IDE, and it looks like a good start. LGTM, thanks @Loghijiaha !